src/node.cc

```diff
@@ -3184,6 +3194,8 @@ static void CheckIfAllowedInEnv(const char* exe, bool is_env,
     "--v8-pool-size",
     "--zero-fill-buffers",
     "-r",
+    "--standalone-mode",
+    "--upstream-node-mode",
```

Comment:
Just want to be clear... are these mutually exclusive modes? Is there ever a scenario where someone would reasonably use both? If not, combining into a single flag with an argument may be better. e.g. `--run-mode=standalone` or `--run-mode=upstream-node`

Reply:
^ https://github.com/nodejs/node/pull/21341#discussion_r195821177

Reply:
It doesn’t feel like CLI flags are quite the right choice for these features, since this only targets embedders, right?

Reply:
@addaleax good question, yes it would only target embedders but i will definitely admit i don't  feel strongly one way or another about how to make it more configurable for upstream :) This is our [commit as it stands](https://github.com/electron/node/commit/7f53cb9ad33b054c7fa226de7042983c104c0331) in our fork

Reply:
@codebytere Thanks, that original context seems helpful here.

Let me try to see if I’m understanding this right: Electron would like … 

- A flag to tell Node.js that it’s not supposed to set up Isolate-wide error handling (`standalone_mode` here). This seems like a good idea to have … I’d suggest this:
  - Move the `isolate->SetPromiseRejectCallback(PromiseRejectCallback);` line into `NewIsolate()`
  - Add an option to `NewIsolate()` that disables these error handling things
- Another flag (`upstream_node_mode`) that … controls the behaviour of `Init()` in some way? i.e. tells it that Node.js is not in control of certain properties of the current process?
  - Is not calling `Init()` at all an option for Electron in that case?
  - Can we get a clearer idea of what exactly the differences are that Electron likes?
  - Either way, it seems like it might be easier to add another parameter to `Init()` as well

The only thing we can’t cover yet this way is the `SetFatalErrorHandler` call, but I’m not sure why Electron does that … it could also be covered through a parameter to `NewIsolate()`, though?